### PR TITLE
[git] switch subcommand

### PR DESCRIPTION
**Pull request by Monkeyank**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
`switch` subcommand for `git`

**What is the current behavior? (You can also link to an open issue here)**
#351

### CODE REVIEW

File: dev/git.ts [hunk 1]
```diff
@@ -4592,6 +4592,149 @@ export const completionSpec: Fig.Spec = {
...
+      name: "switch",
+      description: "Switch branches",
+      options: [
+        {
+          name: ["-c", "--create"],
+          description:
+            "Create a new branch named <new-branch> starting at <start-point> before switching to the branch",
+          args: {
+            name: "new branch",
+          },
```
Comment (QuiiBz) [inline dev/git.ts:4605]: Based on the description, it seems like there is another argument named `start-point`

File: dev/git.ts [hunk 2]
```diff
@@ -4592,6 +4592,149 @@ export const completionSpec: Fig.Spec = {
...
+        },
+        {
+          name: ["-C", "--force-create"],
+          description:
+            "Similar to --create except that if <new-branch> already exists, it will be reset to <start-point>",
+          args: {
+            name: "new branch",
+          },
```
Comment (QuiiBz) [inline dev/git.ts:4613]: Same as above

File: dev/git.ts [hunk 3]
```diff
@@ -4592,6 +4592,149 @@ export const completionSpec: Fig.Spec = {
...
+        },
+        {
+          name: ["-d", "--detach"],
+          description:
+            "Switch to a commit for inspection and discardable experiments",
+        },
+        {
+          name: ["--guess", "--no-guess"],
+          description:
+            "If <branch> is not found but there does exist a tracking branch in exactly one remote (call it <remote>) with a matching name",
```
Comment (QuiiBz) [inline dev/git.ts:4623]: Same as about for `branch`

File: dev/git.ts [hunk 4]
```diff
@@ -4592,6 +4592,149 @@ export const completionSpec: Fig.Spec = {
...
+        },
+        {
+          name: ["-f", "--force"],
+          description: "An alias for --discard-changes",
+        },
+        {
+          name: "--discard-changes",
+          description:
+            "Proceed even if the index or the working tree differs from HEAD. Both the index and working tree are restored to match the switching target",
```
Comment (QuiiBz) [inline dev/git.ts:4632]: These options can have an `isDangerous` option

**Review by QuiiBz — CHANGES_REQUESTED** [review]

File: dev/git.ts (hunk 3 above)
Comment (Monkeyank) [reply dev/git.ts:4623]: I don't think this actually has `args`, but guesses the what the `branch-name` is supposed to be

**Comment by QuiiBz** [comment]
Could you resolve the conflict so we can get this merged?

**Review by QuiiBz — APPROVED** [review]
LGTM!

**Pull request merged**